dotcom-rendering/src/web/components/KeyEventsCarousel.importable.tsx

```diff
@@ -0,0 +1,138 @@
+import { css } from '@emotion/react';
+import type { ArticleFormat } from '@guardian/libs';
+import { from, space } from '@guardian/source-foundations';
+import {
+	Button,
+	buttonThemeBrandAlt,
+	Hide,
+	SvgChevronLeftSingle,
+	SvgChevronRightSingle,
+} from '@guardian/source-react-components';
+import { decidePalette } from '../lib/decidePalette';
+import { KeyEventCard } from './KeyEventCard';
+
+interface Props {
+	keyEvents: Block[];
+	filterKeyEvents: boolean;
+	format: ArticleFormat;
+}
+
+const carouselStyles = (palette: Palette) => css`
+	background-color: ${palette.background.keyEvent};
+	${from.desktop} {
+		background-color: ${palette.background.keyEventFromDesktop};
+	}
+
+	scroll-snap-type: x mandatory;
+	scroll-behavior: smooth;
+	overflow-x: auto;
+	overflow-y: hidden;
+	display: flex;
+	flex-direction: column;
+	scrollbar-width: none; /* for Firefox */
+	-ms-overflow-style: none; /* for Internet Explorer, Edge */
+	&::-webkit-scrollbar {
+		display: none; /* for Chrome, Safari, and Opera */
+	}
+`;
+
+const containerStyles = css`
+	display: flex;
+	justify-content: space-between;
+	flex-direction: row;
+	align-items: stretch;
+	width: fit-content;
+	margin-bottom: ${space[12]}px;
+	position: relative;
+`;
+
+const buttonStyles = css`
+	position: absolute;
+	bottom: 0;
+	margin-bottom: ${space[4]}px;
+	background-color: ${buttonThemeBrandAlt.button.backgroundPrimary};
+	&:active,
+	&:hover {
+		outline: none;
+		background-color: ${buttonThemeBrandAlt.button.backgroundTertiaryHover};
+	}
+`;
+
+const leftButton = css`
+	left: ${space[5]}px;
+`;
+
+const rightButton = css`
+	right: ${space[5]}px;
+`;
+
+const isServer = typeof window === 'undefined';
+const carousel: HTMLElement | null = !isServer
+	? window.document.getElementById('key-event-carousel')
+	: null;
```

Comment:
Can this be achieved with [refs/useRef hook](https://reactjs.org/docs/hooks-reference.html#useref)?

Reply:
Refs hurt my brains. @OllysCoding is it true that the advantage of a ref is you can be sure that `carousel` is always pointing at the right instance of the DOM in the event that there are re-renders? If so, if we don't have state here then do we need to use one?

Reply:
Hmm, actually this section of the docs is quite useful

> However, useRef() is useful for more than the ref attribute. It’s [handy for keeping any mutable value around](https://reactjs.org/docs/hooks-faq.html#is-there-something-like-instance-variables) similar to how you’d use instance fields in classes.
>
> This works because useRef() creates a plain JavaScript object. The only difference between useRef() and creating a {current: ...} object yourself is that useRef will give you the same ref object on every render.

So the question is. Does this component re-render (possible) and does this cause problems for the code using `carousel`?